starter_code/src/movies.js

```diff
@@ -1,15 +1,373 @@
+    title: "The Matrix",
+    year: 1999,
+    director: "Lana Wachowski",
+    duration: "2h 16min",
+    genre: ["Action", "Sci-Fi"],
+    rate: 8.7
+  },
+  {
+    title: "Shichinin no samurai",
+    year: 1954,
+    director: "Akira Kurosawa",
+    duration: "3h 27min",
+    genre: ["Adventure", "Drama"],
+    rate: 8.7
+  },
+  {
+    title: "Star Wars",
+    year: 1977,
+    director: "George Lucas",
+    duration: "2h 1min",
+    genre: ["Action", "Adventure", "Fantasy", "Sci-Fi"],
+    rate: 8.7
+  },
+  {
+    title: "Cidade de Deus",
+    year: 2002,
+    director: "Fernando Meirelles",
+    duration: "2h 10min",
+    genre: ["Crime", "Drama"],
+    rate: 8.7
+  },
+  {
+    title: "Se7en",
+    year: 1995,
+    director: "David Fincher",
+    duration: "2h 7min",
+    genre: ["Crime", "Drama", "Mystery", "Thriller"],
+    rate: 8.6
+  },
+  {
+    title: "The Silence of the Lambs",
+    year: 1991,
+    director: "Jonathan Demme",
+    duration: "1h 58min",
+    genre: ["Crime", "Drama", "Thriller"],
+    rate: 8.6
+  },
+  {
+    title: 'It"s a Wonderful Life',
+    year: 1946,
+    director: "Frank Capra",
+    duration: "2h 10min",
+    genre: ["Drama", "Family", "Fantasy"],
+    rate: 8.6
+  },
+  {
+    title: "La vita è bella",
+    year: 1997,
+    director: "Roberto Benigni",
+    duration: "1h 56min",
+    genre: ["Comedy", "Drama", "War"],
+    rate: 8.6
+  },
+  {
+    title: "The Usual Suspects",
+    year: 1995,
+    director: "Bryan Singer",
+    duration: "1h 46min",
+    genre: ["Crime", "Drama", "Mystery", "Thriller"],
+    rate: 8.6
+  },
+  {
+    title: "Léon",
+    year: 1988,
+    director: "Luc Besson",
+    duration: "1h 50min",
+    genre: ["Crime", "Drama", "Thriller"],
+    rate: 8.6
+  }
+];
+
+console.log("----------------------------");
+
 // Iteration 1: Ordering by year - Order by year, ascending (in growing order)
+function orderByYear(arr) {
+  if (arr.length == 0) {
+    return [null];
+  } else {
+    arr.sort(function(a, b) {
+      return a.year == b.year ? (a.title > b.title) - 0.5 : a.year - b.year;
+    });
+    return arr;
+  }
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
 
+function howManyMovies(arr) {
+  let arr2 = [];
+  let arr3 = [];
+  if (arr.length === 0) {
+    return 0;
+  } else {
+    arr2 = arr.filter(a => a.director == "Steven Spielberg");
+    arr3 = arr2.filter(a => a.genre.indexOf("Drama") !== -1);
+    return arr3.length;
+  }
+}
+
 // Iteration 3: Alphabetic Order - Order by title and print the first 20 titles
+function orderAlphabetically(arrNotMutated) {
+  arrTemp = [...arrNotMutated];
+  if (arrTemp.length == 0) {
+    return [null];
+  } else {
+    arrTemp.sort(function(a, b) {
+      return (a.title > b.title) - 0.5;
+    });
+
+    arrTemp.length >= 20 ? (arrTemp = arrTemp.slice(0, 20)) : arrTemp;
+
+    var arrFinal = [];
+
+    arrTemp.forEach(v => arrFinal.push(v.title));
+
+    return arrFinal;
+  }
+}
```

Comment:
For this function, since the sort method mutates the original array and you must create a new array containing only the titles, you could first  .map() the array (which creates a new array), then sort it then slice it.
I assume that you're comparing the length before slicing in case of some error, if the 2nd parameter is bigger than the arrays length, it will slice until the end.
Solution: 
```
function orderAlphabetically(arrNotMutated) {
  return arrNotMutated.map(m => m.title).sort((a,b) => a.localeCompare(b)).slice(0,20)
}
```